integration/container/health_test.go

```diff
@@ -45,6 +47,44 @@ func TestHealthCheckWorkdir(t *testing.T) {
 	poll.WaitOn(t, pollForHealthStatus(ctx, client, c.ID, types.Healthy), poll.WithDelay(100*time.Millisecond))
 }
 
+func TestHealthCheckFilter(t *testing.T) {
```

Comment:
This change seems like it could be easily tested with a unit test of `transform()` instead of an integration test.